### PR TITLE
Check favicon links against img-src directive

### PR DESCRIPTION
**Pull request by hatstand**

### CODE REVIEW

File: csp_test.go [hunk 1]
```diff
@@ -180,6 +180,18 @@ func TestCSP(t *testing.T) {
 			html:   `<img src="https://blah.com" />`,
 			valid:  false,
 		},
+		{
+			policy: "img-src 'self'",
+			page:   "https://google.com",
+			html:   `<link rel="icon" href="https://blah.com" />`,
+			valid:  false,
```
Comment (d4l3k) [inline csp_test.go:187]: Would be nice to have a positive case here as well

File: csp_test.go [hunk 2]
```diff
@@ -180,6 +180,18 @@ func TestCSP(t *testing.T) {
...
+		},
+		{
+			policy: "img-src 'self'",
+			page:   "https://google.com",
+			html:   `<link rel="apple-touch-icon" href="https://blah.com" />`,
+			valid:  false,
```
Comment (d4l3k) [inline csp_test.go:199]: same here

**Comment by d4l3k** [comment]
Looks good to me! Would be nice if there was a positive case for the tests though

File: csp_test.go (hunk 1 above)
Comment (hatstand) [reply csp_test.go:187]: Done

File: csp_test.go (hunk 2 above)
Comment (hatstand) [reply csp_test.go:199]: Done

**Comment by d4l3k** [comment]
@hatstand Nice, thanks again for contributing!

**Pull request merged**